### ISSUE TITLE
Refactor Product logic

### ISSUE THREAD

**Issue by ntlz** [devops-wishlist-buddies/wishlists#53]
Our products are not real products in the inventory. We should not be concerned about the fact whether or not the product exists in the inventory. We keep our own records of items, so every time we want to add a product to a wishlist, we need to do a POST /wishlists/<wishlist_id>/products  with a payload that includes name, price, etc. To modify a product that is already in a wishlist we would do PUT /wishlists/<wishlist_id>/products/<product_id> .
Following the above, we will have duplicates in items. For example, if user A and user B both have ‘iPhone 12’ in their wishlists, then we will have 2 entries in the database, one item with (belongs to) wishlist_id A, and the other - wishlist_id B. 

TLDR;
Endpoints to modify:
- Create a product in a wishlist, change PUT to POST: `POST /wishlists/<wishlist_id>/products`, with payload `{name: ,
price: , status: , pic_url: , short_desc: , wishlist_id:, inventory_product_id: }`, all fields required. Here we create a new entry in the Products table that belongs to `wishlist_id`.
- Edit an existing product within a wishlist: `PUT /wishlists/<wishlist_id>/products/<product_id>` with a payload  `{name: ,
price: , status: , pic_url: , short_desc: ,wishlist_id:, inventory_product_id: }`, all fields are optional, depending on what needs updating.
